bonsai/BudgetTransactions/BudgetTransactions.swift

```diff
@@ -0,0 +1,175 @@
+//
+
+import SwiftUI
+
+struct VisualEffectView: UIViewRepresentable {
+    var effect: UIVisualEffect?
+    func makeUIView(context: UIViewRepresentableContext<Self>) -> UIVisualEffectView { UIVisualEffectView() }
+    func updateUIView(_ uiView: UIVisualEffectView, context: UIViewRepresentableContext<Self>) { uiView.effect = effect }
+}
+
+struct RoundedCorner: Shape {
+
+    var radius: CGFloat = .infinity
+    var corners: UIRectCorner = .allCorners
+
+    func path(in rect: CGRect) -> Path {
+        let path = UIBezierPath(roundedRect: rect, byRoundingCorners: corners, cornerRadii: CGSize(width: radius, height: radius))
+        return Path(path.cgPath)
+    }
+}
+
+extension View {
+    func cornerRadius(_ radius: CGFloat, corners: UIRectCorner) -> some View {
+        clipShape(RoundedCorner(radius: radius, corners: corners))
+    }
+}
+
+
+struct BudgetTransactions: View {
+    
+    private typealias TransactionGroup = [String: [Transaction]]
+    private let transactionsByDate: TransactionGroup
+    
+    init(transactions: [Transaction]) {
+        var dict: TransactionGroup = [:]
+        for transaction in transactions {
+            let date = Date.dateMapper(date: transaction.date)
+            if dict.keys.contains(date) {
+                if var arr = dict[date] {
+                    arr.append(transaction)
+                    dict[date] = arr
+                }
+            } else {
+                dict[date] = [transaction]
+            }
+        }
+        transactionsByDate = dict
+    }
+        
+    var body: some View {
+        ZStack {
+            VisualEffectView(
+                effect: UIBlurEffect(style: .dark))
+                .edgesIgnoringSafeArea(.all)
+            
+            VStack(alignment: .leading) {
+                HStack {
+                    Spacer()
+                    RoundedCorner()
+                        .foregroundColor(BonsaiColor.disabled)
+                        .frame(width: 60, height: 5, alignment: .center)
+                        .cornerRadius(25, corners: .allCorners)
+                    Spacer()
+                }
+                
+                Text("Transactions")
+                    .font(BonsaiFont.title_headline_17)
+                    .foregroundColor(Color.white)
+                    .padding(.leading, 16)
+                    .frame(height: 17)
+                    .padding(.top, 7)
+    
+                List {
+                    ForEach(transactionsByDate.keys.sorted(by: >), id: \.self) { key in
+                        Section(header: HStack() {
+                            Spacer()
+                            Text("\(key)")
+                                .font(BonsaiFont.body_15)
+                                .foregroundColor(Color.white)
+                            Spacer()
+                        }) {
+                            ForEach(
+                                transactionsByDate[key]!, id: \.self) {
```

Comment:
instead of `ForEach(transactionsByDate.keys)` you could do `ForEach(transactionsByDate)` to get both key and value and therefore you will not need to force unwrap here

Reply:
some syntax mistake, dont know how to satisfied compiler

Reply:
just checked, yep, a problem is that OrderedDictionary is not a RandomAccessCollection
I suggest to fix force unwrap with `transactionByDate[key] ?? []` then, should work just fine

Reply:
fixed)